### PR TITLE
Add functionality to create, update, and delete issue comments

### PR DESCRIPTION
**Pull request by RokasVaitkevicius**
## Type
enhancement

___

## Description
This PR introduces the functionality to create, update, and delete issue comments in the application. It includes the following changes:

- Added `DeleteIssueCommentInput` and `MutationDeleteIssueCommentArgs` types to support deleting issue comments.
- Updated the `IssueComments` component to include UI components and functionality for creating, updating, and deleting issue comments.
- Added an `onUpdateCallback` prop to the `Editor` component to handle editor content updates.
- Added a new UI component, `IssueComment`, for rendering and editing individual issue comments.
- Updated the `IssueDescription` component to use the `onUpdateCallback` prop and the updated mutation for creating issue comments.
- Added new GraphQL mutations (`UPDATE_ISSUE_COMMENT_MUTATION` and `DELETE_ISSUE_COMMENT_MUTATION`) for updating and deleting issue comments.
- Updated the `hocuspocusServer` to handle the `commentRaw` field in issue comments.
- Added new input types (`UpdateIssueCommentInput` and `DeleteIssueCommentInput`) in the GraphQL schema to support updating and deleting issue comments.

___

## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>9 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>graphql.ts&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        frontend/gql/__generated__/graphql.ts<br><br>

**Added `DeleteIssueCommentInput` and <br>`MutationDeleteIssueCommentArgs` types to support deleting <br>issue comments.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/openpro-io/openpro/pull/18/files#diff-c4951772df604fcac415ac7fb71442bb793c5ea43f55e8395a620172a88166f8"> +41/-0</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>IssueComments.tsx&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        frontend/components/IssueModal/IssueComments.tsx<br><br>

**- Added functionality to create, update, and delete issue <br>comments.
- Added UI components for creating, updating, and <br>deleting issue comments.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/openpro-io/openpro/pull/18/files#diff-099c8d5059243da2c8689507e277ed3b1004ca0be61bbcf056edddfee801a76e"> +159/-40</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.tsx&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        frontend/components/Editor/index.tsx<br><br>

**- Added `onUpdateCallback` prop to handle editor content <br>updates.
- Updated `Editor` component to use the <br>`onUpdateCallback` prop.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/openpro-io/openpro/pull/18/files#diff-e3b2076fdaac90c5757c69865802e5c6a017c4aeb52d81887d417d5189519f79"> +18/-16</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>IssueComment.tsx&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        frontend/components/IssueModal/IssueComment.tsx<br><br>

**- Added UI component for rendering and editing individual <br>issue comments.
- Added functionality to update and delete <br>issue comments.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/openpro-io/openpro/pull/18/files#diff-2088c999b627cad2fad00aef41fc522545d0ac8ebfd6b16b7b05a39300df06bf"> +112/-0</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>IssueDescription.tsx&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        frontend/components/IssueModal/IssueDescription.tsx<br><br>

**- Updated `Editor` component to use `onUpdateCallback` <br>prop.
- Updated mutation to use `comment` instead of <br>`defaultContent`.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/openpro-io/openpro/pull/18/files#diff-ab2f001917e3bba42335d7587b51922ab4e1c6ccd3db7767fa73db7ba5f34f99"> +9/-8</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gql-queries-mutations.ts&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        frontend/gql/gql-queries-mutations.ts<br><br>

**- Added `UPDATE_ISSUE_COMMENT_MUTATION` and <br>`DELETE_ISSUE_COMMENT_MUTATION` GraphQL mutations.
- Updated <br>`CREATE_ISSUE_COMMENT_MUTATION` to include `commentRaw` <br>field.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/openpro-io/openpro/pull/18/files#diff-74cc0e508e952171ef14d6d52e268be22319e366a64a6b8454f907cf2561df64"> +17/-0</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.js&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        backend/src/resolvers/IssueComment/index.js<br><br>

**- Added resolvers for creating, updating, and deleting issue <br>comments.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/openpro-io/openpro/pull/18/files#diff-f1ab8fe652f07b623fd9d9a06ecb8bd7524ed8b9f84fc63cda194e1091a8fcc9"> +43/-1</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>hocuspocus-server.js&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        backend/src/services/hocuspocus-server.js<br><br>

**- Updated `hocuspocusServer` to handle issue comment <br>commentRaw field.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/openpro-io/openpro/pull/18/files#diff-2d318c9be6a3e74a863032c7903ae71ea27cff075090a4dd990f13d7bbe76fb3"> +5/-1</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>type-defs.js&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        backend/src/type-defs.js<br><br>

**- Added `UpdateIssueCommentInput` and <br>`DeleteIssueCommentInput` input types.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/openpro-io/openpro/pull/18/files#diff-cf2b118350c3a2869d5aed33eb47c54cfe169ab662623739060c4f1e8cff88b6"> +12/-0</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

### CODE REVIEW

File: frontend/components/Editor/index.tsx [hunk 1]
```diff
@@ -122,7 +121,11 @@ const Editor = ({
       }),
     ],
     onUpdate: ({ editor }) => {
-      if (onUpdateCallback) onUpdateCallback(editor.getJSON());
+      if (onUpdateCallback)
+        onUpdateCallback({
+          content: editor.getJSON(),
+          state: Y.encodeStateAsUpdate(provider.document),
+        });
```
[the remark this answers is not in the document]
Comment (RokasVaitkevicius) [reply frontend/components/Editor/index.tsx:128]: lol, what is editorContent?

[the remark this answers is not in the document]
Comment (claygorman) [reply frontend/components/Editor/index.tsx:128]: let’s mark resolved lol

[the remark this answers is not in the document]
Comment (RokasVaitkevicius) [reply frontend/components/Editor/index.tsx:128]: maybe it meant editor?

File: frontend/components/IssueModal/IssueComments.tsx [hunk 2]
```diff
@@ -23,44 +67,119 @@ const IssueComments = ({ issueId }: { issueId?: string }) => {
...
+          variables: { input: { id: selectedIssueId } },
+        },
+      ],
+      variables: {
+        input: {
+          commentId,
+          comment: JSON.stringify(comment),
+        },
+      },
+    });
```
[the remark this answers is not in the document]
Comment (claygorman) [reply frontend/components/IssueModal/IssueComments.tsx:124]: this we can ignore for now

**Pull request merged**